pkgs/applications/networking/browsers/nyxt/default.nix

```diff
@@ -12,36 +22,62 @@ stdenv.mkDerivation rec {
 
   src = lispPackages.nyxt;
 
-  nativeBuildInputs = [ makeWrapper wrapGAppsHook ];
-  gstBuildInputs = with gst_all_1; [
-    gstreamer gst-libav
+  nativeBuildInputs = [ wrapGAppsHook ];
+  buildInputs = [
+    cairo
+    enchant
+    gdk-pixbuf
+    glib
+    glib-networking
+    gobject-introspection
+    gsettings-desktop-schemas
+    gtk3
+    mime-types
+    pango
+  ] ++ (with gst_all_1; [
+    gstreamer
+    gst-libav
     gst-plugins-base
     gst-plugins-good
     gst-plugins-bad
     gst-plugins-ugly
-  ];
-  buildInputs = [
-    glib gdk-pixbuf cairo
-    mime-types pango gtk3
-    glib-networking gsettings-desktop-schemas
-    xclip notify-osd enchant
-  ] ++ gstBuildInputs;
+  ]);
 
-  GST_PLUGIN_SYSTEM_PATH_1_0 = lib.concatMapStringsSep ":" (p: "${p}/lib/gstreamer-1.0") gstBuildInputs;
```

Comment:
Setting `GST_PLUGIN_SYSTEM_PATH_1_0` is handled by `wrapGApps`.  It suffices to put these in `buildInputs`.